Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/sites/Chan420.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Kuroba - *chan browser https://github.com/Adamantcheese/Kuroba/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.github.adamantcheese.chan.core.site.sites;
+
+import androidx.annotation.Nullable;
+
+import com.github.adamantcheese.chan.core.model.Post;
+import com.github.adamantcheese.chan.core.model.orm.Board;
+import com.github.adamantcheese.chan.core.model.orm.Loadable;
+import com.github.adamantcheese.chan.core.site.Site;
+import com.github.adamantcheese.chan.core.site.SiteIcon;
+import com.github.adamantcheese.chan.core.site.common.CommonSite;
+import com.github.adamantcheese.chan.core.site.common.taimaba.TaimabaActions;
+import com.github.adamantcheese.chan.core.site.common.taimaba.TaimabaApi;
+import com.github.adamantcheese.chan.core.site.common.taimaba.TaimabaCommentParser;
+import com.github.adamantcheese.chan.core.site.common.taimaba.TaimabaEndpoints;
+
+import okhttp3.HttpUrl;
+
+public class Chan420 extends CommonSite {
+    public static final CommonSiteUrlHandler URL_HANDLER = new CommonSiteUrlHandler() {
+        @Override
+        public Class<? extends Site> getSiteClass() {
+            return Chan420.class;
+        }
+
+        @Override
+        public HttpUrl getUrl() {
+            return HttpUrl.parse("https://420chan.org/");
+        }
+
+        @Override
+        public String[] getNames() {
+            return new String[]{"420chan"};
+        }
+
+        @Override
+        public String desktopUrl(Loadable loadable, @Nullable Post post) {
+            if (loadable.isCatalogMode()) {
+                return "https://boards.420chan.org/" + loadable.board.code + "/";
+            } else if (loadable.isThreadMode()) {
+                String url = "https://boards.420chan.org/" + loadable.board.code + "/thread/" + loadable.no;
+                if (post != null) {
+                    url += "#" + post.no;
+                }
+                return url;
+            } else {
+                throw new IllegalArgumentException();
+            }
+		}
+    };
+
+    @Override
+    public void setup() {
+        setName("420Chan");
+        setIcon(SiteIcon.fromFavicon(HttpUrl.parse("https://420chan.org/favicon.ico")));
+
+        setBoards(
```

Comment:
Everything is already enumerated in http://api.420chan.org/boards.json; you should implement the boards method when declaring your endpoints class below and make a Chan420BoardsRequest class to parse that json.